### DISCUSSION TITLE
Frequency Measurement

### DISCUSSION THREAD

**Discussion by w9ran** [khoih-prog/MBED_RPI_PICO_TimerInterrupt#9] [Q&A]
This library looks like a great tool for a lot of applications, but my skill level is such that I'm in need of a little jump-start.   I want to accurately measure the frequency of an audio-range signal (200 Hz to 3kHz) as part of an application.   I know the general approach is to condition the signal and generate an interrupt at each zero crossing.  In between, a timer will measure the period and store it into a variable for me to use.  

I'd prefer to use the Arduino MBED Core, but I found a "RPM Measure" example in the library that supports the Earle Philhower RP2040 boards and have compiled it onto a Pico, but it doesn't seem to work right - and in any case I'll need usec not msec resolution.

I'll keep working on understanding it and the TimerInterrupt library but if anyone can direct me to other examples of frequency measurement using the Pico I'd appreciate it.
Bob

**Comment by khoih-prog** [comment]
Hi @w9ran 

I'm sorry I couldn't find time to help you here. Hopefully someone will chip in.

You also can either adapt the example in  [**Frequency measurement on Seeeduino Xiao at 200kHz**](https://forum.arduino.cc/t/frequency-measurement-on-seeeduino-xiao-at-200khz/688929) from SAMD21 to RP2040 or post the issue on **Arduino Forum** for help from many experts there.

Good Luck,

**Reply by w9ran** [reply]
Thanks for taking the time to reply, and I will look at that example.  For anyone coming across this post, I found a very interesting method of measuring period and duty cycle of PWM signals using the Pico PIO statemachine to do the actual measurement.  In fact it has been updated to measure four signals at the same time:   https://github.com/GitJer/Some_RPI-Pico_stuff/tree/main/PwmIn    

I've converted it to Arduino MBED and have a test program running that I think will do what I need.

Bob
